### PR TITLE
Console colors

### PR DESCRIPTION
**Pull request by kroening**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [X] My contribution is formatted in line with CODING_STANDARD.md.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Comment by kroening** [comment]
This will be hard to test: The effect is only visible when output is _not_ redirected.

File: src/util/cout_message.cpp [hunk 1]
```diff
@@ -35,6 +37,35 @@ cerr_message_handlert::cerr_message_handlert():
...
+    if(SetConsoleMode(out_handle, consoleMode))
+      use_SGR = true;
+  }
+#else
+  is_a_tty = isatty(STDOUT_FILENO);
+  use_SGR = is_a_tty;
+#endif
+}
+
+std::string console_message_handlert::command(unsigned c) const
```
Comment (tautschnig) [inline src/util/cout_message.cpp:66]: Missing documentation. The name does not in an obvious way relate to it's functionality, and what values `c` should take is a mystery.

File: src/util/cout_message.h [hunk 2]
```diff
@@ -36,17 +36,18 @@ class console_message_handlert : public message_handlert
...
-  {
-  }
+  explicit console_message_handlert(bool always_flush);
+
+  std::string command(unsigned c) const override;
 
 protected:
   const bool always_flush;
+  bool is_a_tty;
+  bool use_SGR;
```
Comment (tautschnig) [inline src/util/cout_message.h:54]: ?

File: src/util/message.h [hunk 3]
```diff
@@ -275,6 +280,50 @@ class messaget
     return m;
   }
 
+  std::string command(unsigned c) const
```
Comment (tautschnig) [inline src/util/message.h:286]: Docs.

File: src/util/message.h [hunk 4]
```diff
@@ -275,6 +280,50 @@ class messaget
...
+  {
+    if(message_handler)
+      return message_handler->command(c);
+    else
+      return std::string();
+  }
+
+  // return to default
```
Comment (tautschnig) [inline src/util/message.h]: What is the "default"? Default of what? Also: not Doxygen.

File: src/util/message.h [hunk 5]
```diff
@@ -275,6 +280,50 @@ class messaget
...
+  std::string reset() const
+  {
+    return command(0);
+  }
+
+  // set red foreground color
```
Comment (tautschnig) [inline src/util/message.h]: Doxygen, please.

File: src/util/ui_message.cpp [hunk 6]
```diff
@@ -67,19 +67,24 @@ ui_message_handlert::ui_message_handlert(
   const std::string &program)
   : ui_message_handlert(
       nullptr,
-      cmdline.isset("xml-ui") ? uit::XML_UI : cmdline.isset("json-ui")
-                                                ? uit::JSON_UI
-                                                : uit::PLAIN,
+      cmdline.isset("xml-ui")
+        ? uit::XML_UI
+        : cmdline.isset("json-ui") ? uit::JSON_UI : uit::PLAIN,
```
Comment (tautschnig) [inline src/util/ui_message.cpp]: No change?

File: src/util/ui_message.cpp [hunk 7]
```diff
@@ -67,19 +67,24 @@ ui_message_handlert::ui_message_handlert(
...
-            : cmdline.get_value("timestamp") == "wall"
-                ? timestampert::clockt::WALL_CLOCK
-                : timestampert::clockt::NONE
-        : timestampert::clockt::NONE)
+      cmdline.isset("timestamp") ? cmdline.get_value("timestamp") == "monotonic"
+                                     ? timestampert::clockt::MONOTONIC
+                                     : cmdline.get_value("timestamp") == "wall"
+                                         ? timestampert::clockt::WALL_CLOCK
+                                         : timestampert::clockt::NONE
+                                 : timestampert::clockt::NONE)
```
Comment (tautschnig) [inline src/util/ui_message.cpp]: ?

**Review by tautschnig — CHANGES_REQUESTED** [review]
Essentially items 3 and 7 of the PR checklist. Please don't assign others when those cannot (yet) be ticked as it really means that the focus when reviewing shifts from the meat of the work to those superficial bits.

File: src/util/ui_message.cpp (hunk 7 above)
Comment (kroening) [reply src/util/ui_message.cpp]: clang-format has had fun. Manually reverted.

File: src/util/ui_message.cpp (hunk 6 above)
Comment (kroening) [reply src/util/ui_message.cpp]: Also spurious clang-format change.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 2c09f3b62d1761e089a5cef93f8212738cf3bb8a).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85603358

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 8a40b36ed21260eb8aa151ea48d5943dd44b20ac).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85897347

File: src/util/message.h [hunk 8]
```diff
@@ -291,7 +291,7 @@ class messaget
       return std::string();
   }
 
-  /// return to default formatting
+  /// return to default formatting, as defined by clienta
```
Comment (tautschnig) [inline src/util/message.h]: "clienta" is probably a typo, but I wouldn't understand what the "client" is either.
Nit-pick: Change is in the wrong commit (should just be part of the original commit introducing the documentation).

**Review by tautschnig — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: d999a3e6f9b7fe55f0228e257f3a46dcd3d90f83).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85985276

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**